### DISCUSSION TITLE
prefetchQuery vs refetchQueries

### DISCUSSION THREAD

**Discussion by denisborovikov** [TanStack/query#509] [General] [answered]
Does `refetchQueries` do the same job as `prefetchQuery` if we're speaking about a single query? It seems they're replaceable but have slightly different API.

Updating multiple queries in a single `refetchQueries` call looks weird and bug-prone for me. I feel like `refetchQueries` and `refetchQueries` can be combined into a single function. Where am I mistaken?

**Comment by tannerlinsley** [comment, marked as answer]
`prefetchQuery` will create a query if one does not exist or update the data for one if it does. It's basically like using `useQuery`, but not as a hook

`refetchQueries` is for refetching one (or more) queries that match the query key passed. If none are found, then nothing happens.
